Python/Product/Analysis/Parsing/Parser.cs

```diff
@@ -4197,8 +4211,25 @@ struct Name {
 
         private bool PeekTokenForOrAsyncFor {
             get {
-                return PeekToken(Tokens.KeywordForToken) ||
-                    (AllowAsyncAwaitSyntax && PeekToken(Tokens.KeywordAsyncToken) && PeekToken2()?.Kind == TokenKind.KeywordFor);
+                if (PeekToken(Tokens.KeywordForToken)) {
+                    return true;
+                }
+                if (AllowAsyncAwaitSyntax || (_langVersion >= PythonLanguageVersion.V37 && _inGeneratorExpression)) {
```

Comment:
Should `_langVersion >= PythonLanguageVersion.V37` be simply part of the `AllowAsyncAwaitSyntax` property?

Reply:
No, because it's still not allowed everywhere, but only in generator expressions.